### PR TITLE
added support to set timeout for TAXII requests

### PR DESCRIPTION
**Pull request by joarleymoraes**
Fixing #40

---
This change is [Reviewable](https://reviewable.io/reviews/eclecticiq/cabby/41)

### CODE REVIEW

File: cabby/abstract.py [hunk 1]
```diff
@@ -161,6 +162,7 @@ def _execute_request(self, request, uri=None, service_type=None):
         services by ``service_type``.
         '''
 
+        timeout = self.timeout
```
Comment (traut) [inline cabby/abstract.py]: no need to have a separate local variable

File: cabby/abstract.py [hunk 2]
```diff
@@ -47,6 +47,7 @@ def __init__(self, host=None, discovery_path=None, port=None,
 
         self.jwt_token = None
         self.headers = headers or {}
+        self.timeout = None
```
Comment (traut) [inline cabby/abstract.py]: please add this to constructor arguments

**Comment by traut** [comment]
thank you for the pull request! Timeout is something that should've been added long time ago.

Before I merge it, please adjust your code a bit:
- make sure `flake8` does not complain about `cabby` and `tests` codebases. It shows 2 issues now
- please refactor provided test to use mocked URL with `time.sleep()` call inside. You can use dynamic [body generation](https://github.com/gabrielfalcao/HTTPretty#dynamic-responses-through-callbacks) in httpretty for that. Some users will have network setup that returns an error for `http://httpbin.org:81/` instantly, making the test fail.
- see my comments on the source

**Comment by traut** [comment]
also, please sync your branch with master. I removed `ipdb` dependency and that can help with travis-ci build failure

**Comment by joarleymoraes** [comment]
Ok, will fix the issues.

**Pull request merged**